ledger/apply/keyreg.go

```diff
@@ -49,6 +49,9 @@ func Keyreg(keyreg transactions.KeyregTxnFields, header transactions.Header, bal
 	// (or, if the voting or selection keys are zero, offline/not-participating)
 	record.VoteID = keyreg.VotePK
 	record.SelectionID = keyreg.SelectionPK
+	if balances.ConsensusParams().EnableStateProofKeyregCheck {
+		record.StateProofID = keyreg.StateProofPK
+	}
```

Comment:
nit: why not follow the existing pattern and set `record.StateProofID = merklesignature.Verifier{}` below along with the other fields being cleared out?

Reply:
I thought this was following the existing pattern more faithfully (see the comment above where it describes the change to the Voting and Selection IDs)